### PR TITLE
refactor(app): read robotType from protocol analysis with fallbacks

### PR DESCRIPTION
**Pull request by b-cooper**
# Overview

Remove all instances of the utility function that gleans the robotType from a list of loaded labware
based on soon to be deprecated fixed trash labware loadnames.

The removed function was called `getRobotTypeFromLoadedLabware`

# Review requests

- upload a protocol that references non-slot addressable areas, the deck should render all fixtures, labware, modules, and liquids properly 

# Risk assessment
medium

### CODE REVIEW

File: app/src/organisms/ProtocolDetails/index.tsx [hunk 1]
```diff
@@ -289,6 +289,7 @@ export function ProtocolDetails(
       ? format(new Date(mostRecentAnalysis.createdAt), 'M/d/yy HH:mm')
       : t('shared:no_data')
   const robotType = mostRecentAnalysis?.robotType ?? null
+  console.log('robotType', robotType)
```
Comment (SyntaxColoring) [inline app/src/organisms/ProtocolDetails/index.tsx]: Leftover `console.log()` from debugging?

File: shared-data/js/types.ts [hunk 2]
```diff
@@ -473,6 +473,7 @@ export interface CompletedProtocolAnalysis {
   liquids: Liquid[]
   commands: RunTimeCommand[]
   errors: AnalysisError[]
+  robotType?: RobotType
```
Comment (SyntaxColoring) [inline shared-data/js/types.ts]: Could we type this as `robotType?: RobotType | null` to account for Pydantic limitations that make it easy for the server to accidentally conflate a field being `null` vs. it being omitted? I'm documenting the field as "may be `null` or omitted" on the server side.

**Review by SyntaxColoring — COMMENTED** [review]
HTTP API interface change looks good to me. The server half of this is in #13995.

One `null`-vs-omitted request, though.

File: app/src/organisms/ProtocolDetails/index.tsx (hunk 1 above)
Comment (brenthagen) [inline app/src/organisms/ProtocolDetails/index.tsx]: console.log

Comment (SyntaxColoring) [reply app/src/organisms/ProtocolDetails/index.tsx]: I noticed FIRST >:(

**Review by brenthagen — APPROVED** [review]
goodbye getRobotTypeFromLoadedLabware

**Pull request merged**